app/controllers/discourse_automation/admin_discourse_automation_automations_controller.rb

```diff
@@ -28,15 +28,17 @@ def update
       automation.update!(request.parameters[:automation].slice(:name, :id, :script))
 
       trigger_params = request.parameters[:automation][:trigger].slice(:metadata, :name)
-      if automation.trigger
-        automation.trigger.update_with_params(trigger_params)
-      else
+      unless automation.trigger
```

Comment:
Minor but I think the oneliner reads better

```suggestion
      automation.create_trigger!(trigger_params) unless automation.trigger
```